packages/controllers/src/snaps/SnapController.ts

```diff
@@ -880,7 +889,25 @@ export class SnapController extends BaseController<
    * @param snapId - The id of the Snap whose state to get.
    */
   async getSnapState(snapId: SnapId): Promise<Json> {
-    return this.state.snapStates[snapId] ?? null;
+    const state = this.state.snapStates[snapId];
+    return state ? this.decryptSnapState(snapId, state) : null;
+  }
+
+  private async getEncryptionKey(snapId: SnapId): Promise<string> {
+    return this._getAppKey(`${snapId}_encryption`);
```

Comment:
Why this is not just `this._getAppKey(snapId);`

Reply:
I was hesitant to use a key that is also available to the snap. The snap itself can access the app key for `snapId`, but not for `${snapId}_encryption`.

Reply:
How can Snap access the app key?

Reply:
Using `snap_getAppKey`?